### PR TITLE
async/await convert

### PR DESCRIPTION
**Pull request by Anujraval24**

### CODE REVIEW

**Comment by Anujraval24** [comment]
[Use async / await for all possible functions](https://trello.com/c/biJiamof/2-use-async-await-for-all-possible-functions)

**Comment by Anujraval24** [comment]
[Update export statement for all](https://trello.com/c/45liQEpO/1-update-export-statement-for-all)

**Comment by Anujraval24** [comment]
[Update export statement for all](https://trello.com/c/45liQEpO/1-update-export-statement-for-all)

File: src/extensions/login/login.ts [hunk 1]
```diff
@@ -9,53 +9,55 @@ query UserQuery($email: String!) {
   }
 }`
 
-const getGraphcoolUser = (api: any, email: any) => {
-  return api.request(userQuery, { email })
-    .then(userQueryResult => {
+const login = async (event: any) => {
```
Comment (hardikdot) [inline src/extensions/login/login.ts:12]: why two async function called ?
simple use 
```
export default async (api: any ) => {
  await  api.request(
 );
}
```

File: src/extensions/login/login.ts [hunk 2]
```diff
@@ -9,53 +9,55 @@ query UserQuery($email: String!) {
...
-export = event => {
   if (!event.context.graphcool.pat) {
     console.log('Please provide a valid root token!')
     return { error: 'Email Authentication not configured correctly.' }
   }
 
   // Retrieve payload from event
-  const email = event.data.email
-  const password = event.data.password
+  const { email, password } = event.data
```
Comment (hardikdot) [inline src/extensions/login/login.ts:29]: add validation here

File: src/extensions/login/login.ts (hunk 1 above)
Comment (Anujraval24) [reply src/extensions/login/login.ts:12]: first one is the main function that 
`const login = async () => {
...}`
and second one is to fetch graphcool user

File: src/extensions/login/login.ts (hunk 2 above)
Comment (Anujraval24) [reply src/extensions/login/login.ts:29]:
```
return await getGraphcoolUser(api, email)
    .then((graphcoolUser: any) => {
      if (!graphcoolUser) {
        return Promise.reject('Invalid Credentials')
      } else {
        return bcryptjs.compare(password, graphcoolUser.password)
          .then((passwordCorrect: any) => {
            if (passwordCorrect) {
              let newid = graphcoolUser.id
              console.log('newlogin user id', newid)
              return newid
            } else {
              return Promise.reject('Invalid Credentials')
            }
          })
      }
    })
```

here's the validation

**Pull request closed without merging**